app/models/user.rb

```diff
@@ -1,10 +1,30 @@
 class User < ApplicationRecord
-  has_many :bugs
-  has_many :project_users
-  has_many :projects, through: :project_users
-  enum user_type: { developer: 0, manager: 1, qa: 2 }
   # Include default devise modules. Others available are:
-  #  :lockable, :timeoutable and :omniauthable
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
-end
+
+  enum user_type: { developer: 'developer', manager: 'manager', qa: 'qa' }
+
+  has_many :managed_projects, through: :manager, class_name: 'Project'
+  has_many :created_bugs, class_name: 'Bug', foreign_key: 'creator_id'
+  has_many :assigned_bugs, class_name: 'Bug', foreign_key: 'developer_id'
+  has_many :bugs, foreign_key: :creator_id
```

Comment:
with the has_many :bugs association and the others, where it's specifying the foreign_key as creator_id. If you're following a consistent naming convention, this might not be necessary, as Rails would automatically associate it based on the name :bugs.